internal/services/machinelearning/machine_learning_compute_instance_resource_test.go

```diff
@@ -324,12 +325,13 @@ resource "azurerm_storage_account" "test" {
 }
 
 resource "azurerm_machine_learning_workspace" "test" {
-  name                    = "acctest-MLW%[5]d"
-  location                = azurerm_resource_group.test.location
-  resource_group_name     = azurerm_resource_group.test.name
-  application_insights_id = azurerm_application_insights.test.id
-  key_vault_id            = azurerm_key_vault.test.id
-  storage_account_id      = azurerm_storage_account.test.id
+  name                          = "acctest-MLW%[5]d"
+  location                      = azurerm_resource_group.test.location
+  resource_group_name           = azurerm_resource_group.test.name
+  application_insights_id       = azurerm_application_insights.test.id
+  key_vault_id                  = azurerm_key_vault.test.id
+  storage_account_id            = azurerm_storage_account.test.id
+  public_network_access_enabled = true
```

Comment:
The type should be enum, not boolean, it's not like binary.
It could be many possible values, such as "Enabled", "Disabled", "EnabledForSpecificNetwork", etc.

Reply:
Normally if the enum value only have `enable/disable`, we will convert it to boolean, if the possible values more then two, I will change it back.

Reply:
currently afaik it only supports two values in nearly every resource. if another value is ever add we can always deprecate the property and add non boolean property

Reply:
yes, please.

Reply:
Please keep this to enum value for consistency with the Azure APIs

Reply:
@katbyte   @deeikele and @mingweihe are from MLW service team, they suggest we should keep the enum type consistecy with the AzureAPI and a potential type `EnabledForSpecificNetwork` will be added in  the future.

Reply:
> @katbyte @deeikele and @mingweihe are from MLW service team, they suggest we should keep the enum type consistecy with the AzureAPI and a potential type `EnabledForSpecificNetwork` will be added in the future.

Exactly, I also synced up with the team, they mentioned even though the swagger's still having two values for the properties, the new value will be added later.

I can understand terraform has their own code change policy, while this is really a particular case, keeping a mismatched type there is not suitable for future maintenance.

Reply:
To amplify Mingwei's response boolean type is not appropriate here. Reference to enum implementation is found in Iot Hub (https://learn.microsoft.com/en-us/azure/iot-hub/iot-hub-public-network-access) `selected IP ranges` accepted value.